src/Interpreters/Context.cpp

```diff
@@ -91,6 +93,22 @@ namespace CurrentMetrics
     extern const Metric BackgroundMessageBrokerSchedulePoolTask;
 }
 
+namespace
+{
+
+DB::SessionLogElement makeSessionLogElement(DB::SessionLogElementType type, DB::UserPtr user, const DB::ClientInfo & client_info)
+{
+    DB::SessionLogElement result(type, user->getName());
+    result.client_info = client_info;
+    result.user = user->getName();
+    result.user_identified_with = user->authentication.getType();
+    result.external_auth_server = user->authentication.getLDAPServerName();
```

Comment:
`Context.cpp` is already quite a big file. I mean it's better to move more the code to like that to `SessionLog.h`